libraries/node-core-library/src/FileError.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import * as NodeJSPath from 'path';
+
+import { Path } from './Path';
+import { TypeUuid } from './TypeUuid';
+
+/**
+ * The format that the FileError message should conform to. The supported formats are:
+ *  - Unix: \<filePath\>:\<line\>:\<column\> - \<message\>
+ *  - VisualStudio: \<filePath\>(\<line\>,\<column\>) - \<message\>
+ *  - AzureDevOps: ##vso[task.logissue type=\<error|warning\>;sourcepath=\<filePath\>;linenumber=\<line\>;columnnumber=\<column\>;]\<message\>
+ *
+ * @public
+ */
+export type FileErrorFormat = 'Unix' | 'VisualStudio' | 'AzureDevOps';
+
+/**
+ * Provides options for the output message of a file error.
+ *
+ * @public
+ */
+export interface IFileErrorFormattingOptions {
+  /**
+   * The format for the error message. If no format is provided, format 'Unix' is used by default.
+   *
+   * @remarks If no format is specified and the environment variable TF_BUILD is set to 'True'
+   * (such as on Azure DevOps pipeline agents), the format 'AzureDevOps' is used by default.
+   */
+  format?: FileErrorFormat;
+
+  /**
+   * Whether or not the error is a warning. Defaults to false.
+   */
+  isWarning?: boolean;
+}
+
+const uuidFileError: string = '37a4c772-2dc8-4c66-89ae-262f8cc1f0c1';
+
+/**
+ * An `Error` subclass that should be thrown to report an unexpected state that specifically references
+ * a location in a file.
```

Comment:
Add some `@remarks` clarifying that this class refers to a file that physically exists on disk.

For example, `Could not find a file called "../package.json"` cannot be reported using `FileError`.